packages/loaders/src/tiff/lib/utils.ts

```diff
@@ -137,6 +137,20 @@ function getMultiTiffShapeMap(tiffs: MultiTiffImage[]): {
   };
 }
 
+// If a channel has multiple z or t slices with different samples per pixel
+// this function will just use the samples per pixel from a random slice.
+function getChannelSamplesPerPixel(
+  tiffs: MultiTiffImage[],
+  numChannels: number
+): number[] {
+  const channelSamplesPerPixel = Array(numChannels).fill(0);
+  for (const tiff of tiffs) {
+    const curChannel = tiff.selection.c;
+    channelSamplesPerPixel[curChannel] = tiff.tiff.getSamplesPerPixel();
+  }
+  return channelSamplesPerPixel;
+}
+
```

Comment:
this sort of thing seems like it would indicate an error, but my microscopy knowledge is limited.  how could this happen in practice?

Reply:
i.e different samples per pixel

Reply:
Good catch. It would indicate an error. I'll raise one if it's encountered.